lib/cashu/secret.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  type NUT10SecretData,
+  type ParsedNUT10Secret,
+  type ProofSecret,
+  WELL_KNOWN_SECRET_KINDS,
+} from './types';
+
+/**
+ * Parse secret string from Proof.secret into a well-known secret [NUT-10](https://github.com/cashubtc/nuts/blob/main/10.md)
+ * or a string [NUT-00](https://github.com/cashubtc/nuts/blob/main/00.md)
```

Comment:
I would also maybe mention here that it throws if the secret is nut 10 secret in invalid format

Reply:
done